Dockerfile

```diff
@@ -5,7 +5,10 @@ RUN git config --global --add safe.directory /build
 COPY . .
 RUN make build
 
-FROM builder as test
+FROM builder AS test
+USER 0
+RUN curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/HEAD/install.sh | sh -s -- -b /bin v$(cat .golangciversion)
```

Comment:
Do they by chance have a static bin we could use?

Reply:
nit: Maybe we could even pin the sha of the image

Reply:
`golangci/golangci-lint` is from dockerhub, it may failed to pull in runtime, we should install binary instead https://golangci-lint.run/welcome/install/#binaries

Reply:
Hmm i remember in the past we had a similar discussion and were in favor of copy --from docker image instead? Is that different now?

Reply:
for build container images it maybe ok as we only run them when bump version, but for a app repo require frequent run on pr, this may hit rate limit issue, as https://hub.docker.com/r/golangci/golangci-lint is neither Verified Publisher or Sponsored OSS